pcsx2/GS/Renderers/SW/GSDrawScanlineCodeGenerator.all.cpp

```diff
@@ -1188,22 +1188,35 @@ void GSDrawScanlineCodeGenerator2::TestZ(const XYm& temp1, const XYm& temp2)
 		if (m_sel.zoverflow)
 		{
 			// zs = (GSVector4i(z * 0.5f) << 1) | (GSVector4i(z) & GSVector4i::x00000001());
+			/*GSVector8 z = GSVector8::broadcast32(&scan.p.z) + zo;
+			z /= 2;
+			zs = GSVector8i(z, true);
+			zs = zs.min_u32(GSVector8i::x7fffffff());
+			zs = zs.sll32(1) | 1;*/
 
 			auto m_half = loadAddress(rax, &GSVector4::m_half);
+			auto m_one = loadAddress(rax, &GSVector4i::m_one);
```

Comment:
`loadAddress`  uses the given register, so doing it with the same register multiple times will result in all three being the address of `m_half_int_max`

This does explain the thing that only gets fixed on x64, on 32-bit `rax` isn't used as all addresses are in range, so 32-bit doesn't have any issues here.

You can get yourself constants like `7fffffff`, `80000000`, and `1` by using `cmpeqd` to get all ones, then shifting left/right

Reply:
Hmm that's a good point, I'll change that.